src/hotspot/share/gc/shared/pretouchTask.cpp

```diff
@@ -68,12 +68,6 @@ void PretouchTask::pretouch(const char* task_name, char* start_address, char* en
   // Page-align the chunk size, so if start_address is also page-aligned (as
   // is common) then there won't be any pages shared by multiple chunks.
   size_t chunk_size = align_down_bounded(PretouchTask::chunk_size(), page_size);
-#ifdef LINUX
-  // When using THP we need to always pre-touch using small pages as the OS will
-  // initially always use small pages.
-  page_size = UseTransparentHugePages ? (size_t)os::vm_page_size() : page_size;
-#endif
```

Comment:
It was also the wrong place for this fix since it left out naked calls to os::pretouch_memory.

Reply:
ZGC, and I think Shenandoah, have special handling of UseTransparentHugePages, arranging to similarly
call os::pretouch_memory with small page sizes when that option is true.  It might be that this change
enables some further simplifications (that should perhaps be done as followups rather than added to this).